gui_scripts/gui_helpers/general_helpers.py

```diff
@@ -202,3 +204,210 @@ def stop(self):
             self.paused = False
             self.output_hints_signal.emit('Stopping simulation from thread')
         self.pause_condition.wakeOne()
+
+
+class DirectoryTreeView(QtWidgets.QTreeView):
+    """
+    Sets up a new directory tree view.
+    """
+    item_double_clicked_sig = QtCore.pyqtSignal(QtCore.QModelIndex)
+
+    def __init__(
+        self,
+        file_model: QtWidgets.QFileSystemModel,
+        parent: QtWidgets.QWidget = None
+    ):
+        super().__init__(parent)
+        self.setSelectionBehavior(QtWidgets.QTreeView.SelectRows)
+        self.setSelectionMode(QtWidgets.QTreeView.SingleSelection)
+
+        self.model = file_model
+        self.setModel(self.model)
+
+        self.copied_path = None
+        self.is_directory = False
+        self.is_cut_operation = False
+
+    def copy_item(
+        self,
+        source_index: QtCore.QModelIndex,
+        is_cut_operation: bool = False
+    ):
+        """
+        Implements the copy/cut operations in the directory tree view.
+        :param source_index:        the QModelIndex of the item to be copied
+        or cut.
+        :param is_cut_operation:    boolean to choose between copying
+        and cutting.
+        """
+        # Store the source path and determine if it's a directory
+        self.copied_path = self.model.filePath(source_index)
+        self.is_directory = QtCore.QFileInfo(self.copied_path).isDir()
+        self.is_cut_operation = is_cut_operation
+
+    def copy_directory(
+        self,
+        source_dir: str,
+        destination_dir: str
+    ):
+        """
+        Copies a directory recursively
```

Comment:
I’m curious if this builds properly on sphinx without the period at the end of the description. Would you mind double checking?

Reply:
This is fine